### PR TITLE
Switch to EFI without secure boot

### PR DESCRIPTION
**Pull request by liranr23**
Since we can't transfer NVRAM or TPM data, it is better to have EFI VMs without secure boot.

### CODE REVIEW

**Review by nyoxi — APPROVED** [review]

File: pkg/controller/plan/adapter/ovirt/builder.go [hunk 1]
```diff
@@ -380,11 +380,11 @@ func (r *Builder) mapFirmware(vm *model.Workload, cluster *model.Cluster, object
...
-		smmEnabled := true
-		features.SMM = &cnv.FeatureState{
-			Enabled: &smmEnabled,
-		}
-		firmware.Bootloader = &cnv.Bootloader{EFI: &cnv.EFI{}}
+		secureBootEnabled := false
+		firmware.Bootloader = &cnv.Bootloader{
+			EFI: &cnv.EFI{
+				SecureBoot: &secureBootEnabled,
+			}}
```
Comment (ahadas) [inline pkg/controller/plan/adapter/ovirt/builder.go:387]: the changes look good but how about documenting (in the code) why we do that?

Comment (ahadas) [reply pkg/controller/plan/adapter/ovirt/builder.go:387]: hmm this part is actually self-explained, I meant after adding this handling to the other 'bios-type' that represents uefi+secure boot

File: validation/policies/io/konveyor/forklift/vmware/uefi_boot.rego [hunk 2]
```diff
@@ -8,7 +8,7 @@ concerns[flag] {
     has_uefi_boot
     flag := {
         "category": "Warning",
-        "label": "UEFI secure boot detected",
-        "assessment": "UEFI secure boot is enabled. NVRAM data is not copied during the migration and thus the VM might not boot on OpenShift Virtualization."
+        "label": "UEFI detected",
+        "assessment": "If UEFI secure boot is enabled, NVRAM data is not copied during the migration and thus the VM might not boot on OpenShift Virtualization."
```
Comment (nyoxi) [inline validation/policies/io/konveyor/forklift/vmware/uefi_boot.rego:12]: I would explicitly mention that the migtrated VM will have secure boot disabled.

**Review by nyoxi — CHANGES_REQUESTED** [review]
Actually, I don't understand the sentence in commit message: `As for some VMs it may the boot process of it.`

File: validation/policies/io/konveyor/forklift/vmware/uefi_boot.rego (hunk 2 above)
Comment (ahadas) [reply validation/policies/io/konveyor/forklift/vmware/uefi_boot.rego:12]: yeah, maybe something like:
```suggestion
        "assessment": "UEFI secure boot will be disabled on Openshift Vrtualization. If the VM was set with UEFI secure boot, manual steps within the guest would be needed in order to boot the guest operating system."
```

**Review by nyoxi — APPROVED** [review]

File: pkg/controller/plan/adapter/ovirt/builder.go [hunk 3]
```diff
@@ -380,11 +381,12 @@ func (r *Builder) mapFirmware(vm *model.Workload, cluster *model.Cluster, object
 	}
 	switch biosType {
 	case Q35Ovmf:
-		smmEnabled := true
-		features.SMM = &cnv.FeatureState{
-			Enabled: &smmEnabled,
-		}
-		firmware.Bootloader = &cnv.Bootloader{EFI: &cnv.EFI{}}
```
Comment (ahadas) [inline pkg/controller/plan/adapter/ovirt/builder.go:387]: I think you'll need to add `fallthrough` here if you want to switch to the next block or do `case Q35Ovmf, Q35SecureBoot:`

File: validation/policies/io/konveyor/forklift/vmware/uefi_boot.rego [hunk 4]
```diff
@@ -8,7 +8,7 @@ concerns[flag] {
     has_uefi_boot
     flag := {
         "category": "Warning",
-        "label": "UEFI secure boot detected",
-        "assessment": "UEFI secure boot is enabled. NVRAM data is not copied during the migration and thus the VM might not boot on OpenShift Virtualization."
+        "label": "UEFI detected",
+        "assessment": "UEFI secure boot will be disabled on Openshift Vrtualization. If the VM was set with UEFI secure boot, manual steps within the guest would be needed for the guest operating system to boot."
```
Comment (ahadas) [inline validation/policies/io/konveyor/forklift/vmware/uefi_boot.rego:12]:
```suggestion
        "assessment": "UEFI secure boot will be disabled on Openshift Virtualization. If the VM was set with UEFI secure boot, manual steps within the guest would be needed for the guest operating system to boot."
```

**Review by ahadas — APPROVED** [review]

**Pull request merged**